modules/engine/src/lib/pipeline-factory.ts

```diff
@@ -104,9 +104,10 @@ export class PipelineFactory {
     const bufferLayoutHash = this._getHash(JSON.stringify(props.bufferLayout));
 
     switch (this.device.type) {
-      // case 'webgl':
-      // WebGL is more dynamic
-      // return `${vsHash}/${fsHash}V${varyingHash}BL${bufferLayoutHash}`;
+      case 'webgl':
```

Comment:
Not part of this PR, but should we move the hash logic to each device?

```ts
device.getRenderPipelineHash(props: RenderpipelineProps): string
```

Reply:
Yes... Perhaps we could just expose the capabilities instead, perhaps using the device feature system: `dynamic-topology-webgl`, `dynamic-parameters-webgl` so that the PipelineFactory could take those into account without switching on device type, and the factory could still own and encapsulate the hashing system.